sherpa/astro/data.py

```diff
@@ -4192,47 +4192,59 @@ def _fix_y_units(self, val, filter=False, response_id=None):
             areascal = self._check_scale(self.areascal, filter=filter)
             val /= areascal
 
-        if self.grouped or self.rate:
-
+        # Should this be a user-callable method?
+        #
+        def get_bin_edges():
             if self.units != 'channel':
-                elo, ehi = self._get_ebins(response_id, group=False)
+                xlo, xhi = self._get_ebins(response_id, group=False)
             else:
-                elo, ehi = (self.channel, self.channel + 1.)
+                xlo, xhi = (self.channel, self.channel + 1.)
 
             if filter:
                 # If we apply a filter, make sure that
                 # ebins are ungrouped before applying
                 # the filter.
-                elo = self.apply_filter(elo, self._min)
-                ehi = self.apply_filter(ehi, self._max)
+                xlo = self.apply_filter(xlo, self._min)
+                xhi = self.apply_filter(xhi, self._max)
             elif self.grouped:
-                elo = self.apply_grouping(elo, self._min)
-                ehi = self.apply_grouping(ehi, self._max)
+                xlo = self.apply_grouping(xlo, self._min)
+                xhi = self.apply_grouping(xhi, self._max)
 
-            if self.units == 'energy':
-                ebin = ehi - elo
-            elif self.units == 'wavelength':
-                ebin = hc / elo - hc / ehi
-            elif self.units == 'channel':
-                ebin = ehi - elo
+            return xlo, xhi
+
+        if self.grouped or self.rate:
+            xlo, xhi = get_bin_edges()
+            if self.units == 'wavelength':
+                dx = hc / xlo - hc / xhi
             else:
```

Comment:
Suggest added comment for future readers
```suggestion
            else:  # Could be "energy" or "channel"
```